Cython/Utility/ModuleSetupCode.c

```diff
@@ -1037,6 +1051,38 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_SET_SIZE(obj, size) Py_SIZE(obj) = (size)
 #endif
 
+#if CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS && PY_VERSION_HEX >= 0x030d00b1
+#define __Pyx_PyList_GetItemRef(o, i) PyList_GetItemRef(o, i)
+#define __Pyx_PyDict_GetItemRef(dict, key, result) PyDict_GetItemRef(dict, key, result)
+#elif !CYTHON_AVOID_BORROWED_REFS
+
+#if CYTHON_ASSUME_SAFE_MACROS
+#define __Pyx_PyList_GetItemRef(o, i) __Pyx_NewRef(PyList_GET_ITEM(o, i))
+#else
+#define __Pyx_PyList_GetItemRef(o, i) __Pyx_NewRef(PyList_GetItem(o, i))
```

Comment:
`__Pyx_NewRef()` doesn't do any error checking, so if `PyList_GetItem()` fails (for whatever reason, e.g. because it's PyPy and accidentally needs to do memory allocation that fails), then this would probably crash.
```suggestion
#define __Pyx_PyList_GetItemRef(o, i) PySequence_GetItem(o, i)
```